src/node/executor.rs

```diff
@@ -100,53 +91,26 @@ fn spawn_messages_handler(
                         recv,
                     };
 
-                    stage.lock().await.send_event(event);
+                    stage.send_event(event).await;
                 }
             }
         }
     }));
 }
 
-fn spawn_node_message_handler(stage: Arc<Mutex<Stage>>, msg_bytes: Bytes, sender: SocketAddr) {
+fn spawn_node_message_handler(stage: Arc<Stage>, msg_bytes: Bytes, sender: SocketAddr) {
     let _ = tokio::spawn(async move {
         match Message::from_bytes(&msg_bytes) {
-            Ok(msg) => {
-                // Process the message according to our stage
-                let mut stage = stage.lock().await;
-                if let Err(err) = stage.process_message(sender, msg.clone()).await {
-                    error!(
-                        "{}Error encountered when processing message {:?}: {}",
-                        stage.log_ident(),
-                        msg,
-                        err
-                    );
-                }
+            Ok(message) => {
+                let command = Command::HandleMessage {
+                    message,
+                    sender: Some(sender),
+                };
+                let _ = stage.handle_commands(command).await;
```

Comment:
it's actually `handle_command` ?

Reply:
No, `handle_command` (singular) handles only one command and returns a `Vec` of further commands to handle. `handle_commands` (plural) also handles those returned commands (and those returned from them, etc...).